aea/protocols/dialogue/base.py

```diff
@@ -1134,18 +1146,50 @@ def remove(self, dialogue_label: DialogueLabel) -> None:
 
         :param dialogue_label: label of the dialogue to remove
         """
-        dialogue = self._dialogues_by_dialogue_label.pop(dialogue_label, None)
+        if dialogue_label.is_complete():
+            complete_dialogue_label: Optional[DialogueLabel] = dialogue_label
+            incomplete_dialogue_label = dialogue_label.get_incomplete_version()
+        else:
+            complete_dialogue_label = None
+            incomplete_dialogue_label = dialogue_label
 
-        self._incomplete_to_complete_dialogue_labels.pop(dialogue_label, None)
+        # TODO: by design all incomplete dialogue labels must be present in this mapping  # pylint: disable=fixme
```

Comment:
There is some issue with the persisted storage implementation making this not true atm. We need to fix that!

Reply:
Does this mean that we must keep all labels?

Reply:
> If you remove None as a default return and try to run the tests then they fail. We need to fix repopulating some dictionaries after loading from persisted storage